### ISSUE TITLE
Using datashade=True flips the image up-down

### ISSUE THREAD

**Issue by bidhya** [holoviz/datashader#1054]
#### Description of expected behavior and the observed behavior
Using datashade=True option the plotted image is flipped up-down. Is there some bug due to datashade or I am doing something wrong. I also see the dynamic range of color changed as well. 

See the two images below side-by-side, with only change being 'datashade=True'

Here dem is any 1-band DEM
#### Complete, minimal, self-contained example code that reproduces the issue

```
dem = dems[0] 
ds =rioxarray.open_rasterio(f'{base_folder}/{dem}')
print(f'DEM: {dem}')
print(ds.shape)

asp = ds.shape[-1]/ds.shape[1]

ds.hvplot.image(x='x', y='y', aspect=asp, frame_width=250, cmap='gist_earth') + ds.hvplot.image(datashade=True, x='x', y='y', aspect=asp, frame_width=250, cmap='gist_earth')

```
#### Screenshots or screencasts of the bug in action
![hvplot_flip_datashade](https://user-images.githubusercontent.com/6469101/104520894-dd07de80-55c9-11eb-9126-b37e717580fc.png)

#### ALL software version info
all these libraies from conda-forge
hvplot:  0.7.0  
datashader 0.12.0  
holoviews 1.14.1  
xarray 0.16.2  
rioxarray 0.1.1

**Comment by ianthomas23** [comment]
This is indeed a bug in datashader. It is simple to understand, but is going to be awkward to fix without breaking other use cases.

The clue is that GeoTIFF files usually (always?) have their dimensions in the order `(y, x)` and the y-coordinates are decreasing. Other uses still use the numpy standard of `(y, x)` but coordinates increasing, plus it is entirely possible to use `(x, y)` dimensions instead. In `datashader/utils.py` we have inconsistent handling of the dimension order:

- `calc_res` assumes `(y, x)` and flips the y-axis.
- `calc_bbox` assumes `(x, y)`.
- `orient_array` assumes `(y, x)` as most of the work is done by `calc_res`.
- `downsample_aggregate` uses `(x, y)`.

We could do a localised fix in `calc_res` to check the dimension order and behave correctly, but then I suspect that (possibly incorrect) assumptions made elsewhere in datashader and possibly even in HoloViews/hvPlot might be affected.

Various functions calculate and/or pass around array properties named similar to `(xprop, yprop)`. In some cases these might actually be referring to properties in the `(x, y)` order and in others they might be referring to properties in the same order as the array indices.

**Comment by jbednar** [comment]
Fixed in #1095.
